### PR TITLE
cmake: Freshen up docs/INSTALL.cmake

### PR DESCRIPTION
**Pull request by levitte**
- Turn docs/INSTALL.cmake into a proper markdown file, docs/INSTALL.cmake.md
- Move things around to divide the description into configuration, building
  and installing sections
- Mention the more modern cmake options to configure, build and install, but
  also retain the older variants as fallbacks

### CODE REVIEW

File: docs/INSTALL.cmake [hunk 1]
```diff
@@ -41,49 +41,56 @@ Command Line CMake
...
-    $ make
+          $ cmake -B .
 
-    3. Install to default location:
+       b. Build in a separate directory (parallel to the curl source
+          tree in this example).  The build directory will be created
+          for you.
 
-    $ make install
+          $ cmake -B ../curl-build
```
[the remark this answers is not in the document]
Comment (levitte) [reply docs/INSTALL.cmake:57]: Update done

[the remark this answers is not in the document]
Comment (vszakats) [reply docs/INSTALL.cmake:57]: 3.7 was released on 2016-11-11. Not new, but not terribly old. It
was bumped to this about a year ago from 3.2. curl supports tools
older than this one.

I'd discuss a requirement bump first, then if accepted adjust the
docs and other things to the new baseline.

FWIW 3.17 was released on 2019-07-17.

[the remark this answers is not in the document]
Comment (levitte) [reply docs/INSTALL.cmake:57]: Okie.  I'll convert this PR to draft for the time being, just to make it clear it's not to be merged right now

[the remark this answers is not in the document]
Comment (levitte) [reply docs/INSTALL.cmake:57]: I hope I made the right move by creating the discussion #12774

**Review by bagder — COMMENTED** [review]
Maybe change this document to use a `.md` extension so that it will be treated as markdown properly by GitHub, editors etc?

**Comment by levitte** [comment]
> Maybe change this document to use a `.md` extension so that it will be treated as markdown properly by GitHub, editors etc?

Sure.  INSTALL.cmake.md, then?

**Comment by levitte** [comment]
Fixup commit added, for separate review.  I'll squash it into the first commit later

**Comment by levitte** [comment]
I listened to both criticism (the minimum cmake requirement) and ideas (turn it into a proper markdown file) and took a stab at trying to answer them all. I hope that what I've now pushed does the trick.

**Comment by levitte** [comment]
... with that, I can't see this PR being a draft any more.

**Comment by vszakats** [comment]
I'd suggest naming the file `INSTALL-CMAKE.md`. This matches what we use in that directory and avoids the double-extension.

**Comment by levitte** [comment]
> I'd suggest naming the file `INSTALL-CMAKE.md`. This matches what we use in that directory and avoids the double-extension.

Wish granted!

**Review by vszakats — APPROVED** [review]

**Comment by vszakats** [comment]
Thank you @levitte, LGTM!

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**